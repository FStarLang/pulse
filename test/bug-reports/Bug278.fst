module Bug278

#lang-pulse
open Pulse

assume val domain : a:Type -> (a -> slprop) -> Type

assume val spawn :
 (#a:Type) -> (#pre : slprop) -> (#post : (a -> slprop)) ->
 ($f : unit -> stt a pre post) ->
 stt (domain a post) pre (fun _ -> emp)

fn pth (n:nat) (_:unit)
  requires emp
  returns n:nat
  ensures emp
{
  2
}

fn pfib (n:nat)
  requires emp
  ensures emp
{
  ();
  let _n = spawn u#0 u#0 #nat #emp #(fun (n:nat) -> emp) (pth n);
  ();
}

[@@expect_failure] // for some reason we need to provide universes
fn pfib2 (n:nat)
  requires emp
  ensures emp
{
  ();
  let _n = spawn #nat #emp #(fun (n:nat) -> emp) (pth n);
  ();
}
