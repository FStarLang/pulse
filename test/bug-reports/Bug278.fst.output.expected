* Info at Bug278.fst(35,5-37,5):
  - Expected failure:
  - Tactic failed
  - Internal error: unexpected unresolved (universe) uvar in deep_compress: 22

